### DISCUSSION TITLE
compose.yaml: nextcloud-aio-mastercontainer should always be connected to the nextcloud-aio network

### DISCUSSION THREAD

**Discussion by pmorch** [nextcloud/all-in-one#3252] [Questions] [answered]
### Steps to reproduce
1. Create `nextcloud-aio-mastercontainer` with `docker-compose up -d` with network sections commented out as in the template.
2. Create and start containers with
    `docker exec -it --env START_CONTAINERS=1 nextcloud-aio-mastercontainer /daily-backup.sh`
4. See that ping (or subsequent `/daily-backup.sh` commands) fail

### Expected behavior 

```shell
$ sudo docker exec -it nextcloud-aio-mastercontainer ping nextcloud-aio-redis
PING nextcloud-aio-redis (172.31.0.4): 56 data bytes
64 bytes from 172.31.0.4: seq=0 ttl=64 time=0.105 ms
64 bytes from 172.31.0.4: seq=1 ttl=64 time=0.111 ms
64 bytes from 172.31.0.4: seq=2 ttl=64 time=0.123 ms
^C
```

### Actual behavior 

```shell
$ sudo docker exec -it nextcloud-aio-mastercontainer ping nextcloud-aio-redis
ping: bad address 'nextcloud-aio-redis'
```

### Mediation

This mediates it: Turns actual into expected:

```shell
$ sudo docker network connect nextcloud-aio nextcloud-aio-mastercontainer
```

After which ping starts working

#### Creating containers from GUI works

If I create the containers from the AIO GUI, it adds the `nextcloud-aio` network to the `nextcloud-aio-mastercontainer` and ping works immediately.

#### Recreating `nextcloud-aio-mastercontainer`  causes this to re-appear

Regardless of whether I created the containers with the AIO GUI or with `/daily-backup.sh` and then manually added the network afterwards, if I recreate `nextcloud-aio-mastercontainer` with this sequence:

```shell
sudo docker-compose down
sudo docker-compose up -d
```

That will stop and remove `nextcloud-aio-mastercontainer` and recreate it again, this time without the `nextcloud-aio` network. Again this mediates it:

```shell
$ sudo docker network connect nextcloud-aio nextcloud-aio-mastercontainer
```

### Suggested fix

Always add `nextcloud-aio` network to `nextcloud-aio-mastercontainer`. Then it works in all the scenarios I could test.

Which is exactly pull request https://github.com/nextcloud/all-in-one/pull/3251

### Host OS 

Synology 7.1.1

#### Nextcloud AIO version 

```shell
$ sudo docker ps --format '{{.Names}} {{.Image}}' | grep nextcloud
nextcloud-aio-mastercontainer nextcloud/all-in-one:latest
nextcloud-aio-apache nextcloud/aio-apache:latest
nextcloud-aio-notify-push nextcloud/aio-notify-push:latest
nextcloud-aio-nextcloud nextcloud/aio-nextcloud:latest
nextcloud-aio-redis nextcloud/aio-redis:latest
nextcloud-aio-database nextcloud/aio-postgresql:latest
```

**Comment by szaimen** [comment, marked as answer]
Hi, see 
![image](https://github.com/nextcloud/all-in-one/assets/42591237/477a6938-01bc-432d-a101-0b057cb358c0)
